### PR TITLE
[KYUUBI #5377][FOLLOWUP] Get limit from more spark plan and regard result max rows

### PR DESCRIPTION
**Pull request by turboFei**
# :mag: Description
Followup #5591 
Support to get existing limit from more plan and regard the result max rows.
## Issue References 🔗

This pull request fixes #

## Describe Your Solution 🔧

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

**Comment by turboFei** [comment]
@lsm1 @cxzl25 @pan3793

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala [hunk 1]
```diff
@@ -295,13 +294,21 @@ object SparkDatasetHelper extends Logging {
...
+  }
+
   def shouldSaveResultToFs(resultMaxRows: Int, minSize: Long, result: DataFrame): Boolean = {
     if (isCommandExec(result.queryExecution.executedPlan.nodeName)) {
       return false
     }
-    lazy val limit = result.queryExecution.executedPlan match {
-      case collectLimit: CollectLimitExec => collectLimit.limit
-      case _ => resultMaxRows
+    val limit = planLimit(result.queryExecution.executedPlan) match {
```
Comment (ulysses-you) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala:308]: it seems it's more simpler to match a logical limit

**Review by ulysses-you — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala [hunk 2]
```diff
@@ -295,13 +295,19 @@ object SparkDatasetHelper extends Logging {
...
+  }
+
   def shouldSaveResultToFs(resultMaxRows: Int, minSize: Long, result: DataFrame): Boolean = {
     if (isCommandExec(result.queryExecution.executedPlan.nodeName)) {
       return false
     }
-    lazy val limit = result.queryExecution.executedPlan match {
-      case collectLimit: CollectLimitExec => collectLimit.limit
-      case _ => resultMaxRows
+    val limit = logicalPlanLimit(result.queryExecution.logical) match {
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala:307]: can we have a different name for this variable? e.g. `finalLimit`, `effectiveLimit`, to distinguish with the `Some(limit)`

**Review by pan3793 — APPROVED** [review]
LGTM, one minor suggestion

**Review by cfmcgrady — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

**Comment by turboFei** [comment]
logical plan can not cover the case likes:
```
SELECT * FROM (SELECT * FROM tv ORDER BY id LIMIT $topKThreshold) AS TEMP
```

So, I revert to use sparkPlan.

**Comment by ulysses-you** [comment]
@turboFei can we use optimizedPlan ? `SubqueryAlias` should only at analyzed side.

**Comment by turboFei** [comment]
addressed. @ulysses-you

**Review by ulysses-you — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master

**Pull request closed without merging**